api/v1beta1/machinehealthcheck_types.go

```diff
@@ -24,6 +24,27 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+// MachineHealthCheck's RemediationAllowed condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// MachineHealthCheckRemediationAllowedV1Beta2Condition surfaces whether the MachineHealthCheck is
+	// allowed to remediate any Machines or whether it is blocked from remediating any further.
+	MachineHealthCheckRemediationAllowedV1Beta2Condition = "RemediationAllowed"
+
+	// MachineHealthCheckTooManyUnhealthyV1Beta2Reason is the reason used when too many Machines are unhealthy and
+	// the MachineHealthCheck is blocked from making any further remediation.
+	MachineHealthCheckTooManyUnhealthyV1Beta2Reason = "TooManyUnhealthy"
+
+	// MachineHealthCheckRemediationAllowedV1Beta2Reason is the reason used when the number of unhealthy machine
+	// is within the limits defined by the MachineHealthCheck, and thus remediation is allowed.
+	MachineHealthCheckRemediationAllowedV1Beta2Reason = "RemediationAllowed"
+)
+
+// MachineHealthCheck's Paused condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// MachineHealthCheckPausedV1Beta2Condition is true if this MachineHealthCheck or the Cluster it belongs to are paused.
+	MachineHealthCheckPausedV1Beta2Condition = PausedV1Beta2Condition
+)
```

Comment:
I assume not needed anymore after Christian's PR?

(re-opened: https://github.com/kubernetes-sigs/cluster-api/pull/11290/files#r1800968781)

(it's also unused in this PR)

Reply:
Yep, I assume @chrischdi PR will cleanup all the paused conditions if we don't need them anymore (there is one for each object)

Reply:
Noted. Will comment on the paused PR in case it's not already done there